compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageCompilation.java

```diff
@@ -71,7 +71,7 @@ private PackageCompilation(PackageContext rootPackageContext) {
 
     private PackageCompilation(PackageContext rootPackageContext, CompilationOptions compilationOptions) {
         this.rootPackageContext = rootPackageContext;
-        this.packageResolution = rootPackageContext.getResolution(compilationOptions);
+        this.packageResolution = rootPackageContext.getResolution();
```

Comment:
We can refactor this class to remove duplicated code now. Shall we do that improvment?

Reply:
Refactored via https://github.com/ballerina-platform/ballerina-lang/pull/35067/commits/f2736d713095205a010a22682fb483e0c3965d06